src/osd/PrimaryLogPG.cc

```diff
@@ -2517,6 +2517,32 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
   maybe_force_recovery();
 }
 
+bool PrimaryLogPG::need_skip_handle_manifest(const OpRequestRef& op_req,
+                                             const bool is_chunked)
+{
+  // following ops don't need manifest handling
+  for (auto& osd_op : op_req->get_req<MOSDOp>()->ops) {
+    if (osd_op.op.op == CEPH_OSD_OP_SET_REDIRECT ||
+        osd_op.op.op == CEPH_OSD_OP_SET_CHUNK ||
+        osd_op.op.op == CEPH_OSD_OP_UNSET_MANIFEST ||
+        osd_op.op.op == CEPH_OSD_OP_TIER_PROMOTE ||
+        osd_op.op.op == CEPH_OSD_OP_TIER_FLUSH ||
+        osd_op.op.op == CEPH_OSD_OP_TIER_EVICT ||
+        osd_op.op.op == CEPH_OSD_OP_ISDIRTY) {
+      return true;
+    }
+
+    /*  in case of chunked manifest objects, read-only cls ops
+     *  that are not read_data don't need to promote chunk objects.
+     */
+    if (is_chunked && osd_op.op.op == CEPH_OSD_OP_CALL &&
```

Comment:
Why do we need the osd_op.op.op == CEPH_OSD_OP_CALL check?

Reply:
I have only filtered for cls operation because I have not tested other operations specifically. 
I think it will be okay to check all operations generally without this condition.
I'll fix it.